pyannote/audio/core/io.py

```diff
@@ -75,20 +82,33 @@ class Audio:
     >>> audio = Audio(sample_rate=16000, mono=True)
     >>> waveform, sample_rate = audio({"audio": "/path/to/audio.wav"})
     >>> assert sample_rate == 16000
-
-    >>> two_seconds_stereo = np.random.rand(44100 * 2, 2, dtype=np.float32)
-    >>> waveform, sample_rate = audio({"waveform": two_seconds_stereo, "sample_rate": 44100})
+    >>> sample_rate = 41000
+    >>> two_seconds_stereo = torch.rand(2, 2*sample_rate)
+    >>> waveform, sample_rate = audio({"waveform": two_seconds_stereo, "sample_rate": sample_rate})
     >>> assert sample_rate == 16000
     >>> assert waveform.shape[1] == 1
     """
 
     @staticmethod
-    def normalize(waveform: np.ndarray) -> np.ndarray:
-        return waveform / (np.sqrt(np.mean(waveform ** 2)) + 1e-8)
+    def normalize(waveform: Tensor) -> Tensor:
+        """
+
+        Parameters
+        ----------
+        waveform : (channel, time) Tensor
+            Single or multichannel waveform
+
+
+        Returns
+        -------
+        waveform: (channel, time) Tensor
+        """
+        stds = waveform.pow(2).mean(1).sqrt().add(1e-8)
+        return (waveform.permute(1, 0) / stds).permute(1, 0)
 
     @staticmethod
     def get_duration(file: AudioFile) -> float:
```

Comment:
@hbredin This could be a cached property?

Reply:
A property only depends on `self`. This depends on an external `AudioFile`.

This could be a cached _method_ (not _property_) but it would also mean that `AudioFile` instances must be hashable (which I am not sure they are for now, given that an `AudioFile` is basically a `dict` whose keys and values can be basically anything.